services/src/util/apidoc.rs

```diff
@@ -148,19 +161,31 @@ impl Modify for TransformSchemasWithTag {
                 let Some(variant_tag) = Self::get_variant_tag(item, discriminator) else {
                     continue 'outer;
                 };
-                let variant_schema_name = match Self::get_base_type_name(item) {
-                    Some(base_type) => format!(
-                        "{}With{}",
-                        base_type,
-                        Self::uppercase_first_letter(discriminator),
-                    ),
-                    None => format!(
-                        "{}{}",
-                        Self::uppercase_first_letter(variant_tag),
-                        schema_name
-                    ),
+
+                let variant_schema_name = match Self::get_title(item) {
+                    // Always prefer custom title as variant name.
+                    Some(title) => title.to_owned(),
+                    None => match Self::get_base_type_name(item) {
+                        // If the variant has a single tuple payload, try using
+                        // the name unchanged and just add a "type" field to it.
+                        Some(base_type_name) => base_type_name.to_owned(),
+                        // If the autogenerated name sounds bad in some cases, you can
+                        // set #[schema(title = "XXX")] or create a new struct type and
+                        // use it as tuple variant. Then the hardcoded name will be used.
+                        None => format!(
+                            "{}{}",
+                            schema_name,
+                            Self::uppercase_first_letter(variant_tag)
+                        ),
+                    },
                 };
 
+                if matches!(schema_uses.get(&variant_schema_name), Some(count) if count != &1usize)
+                {
+                    panic!("The type {variant_schema_name} is used in the enum {schema_name} as payload, but also in other places. \
```

Comment:
can you add a test on a faulty schema that ensures that this error gets thrown?

Reply:
[c42aa7d](https://github.com/geo-engine/geoengine/pull/953/commits/c42aa7d4d143e8c6c6454744491e427e255bbbf3)